File: articles/behavioral-finance-investor-psychology/c/behavioral_demand.c
#include <stdio.h>

double behavioral_demand(double expected_return, double perceived_variance, double behavioral_term, double alpha, double beta, double gamma) {
    return alpha * expected_return - beta * perceived_variance + gamma * behavioral_term;
}

double net_return_after_turnover(double gross_return, double turnover, double cost_per_turnover) {
    return gross_return - cost_per_turnover * turnover;
}

int main(void) {
    double demand = behavioral_demand(0.08, 0.03, 0.40, 1.2, 0.7, 0.5);
    double net = net_return_after_turnover(0.05, 1.4, 0.0025);
    printf("Synthetic behavioral demand: %.3f\n", demand);
    printf("Synthetic net return after turnover: %.3f\n", net);
    return 0;
}

File: articles/behavioral-finance-investor-psychology/fortran/behavioral_demand.f90
program behavioral_demand
  implicit none
  real :: expected_return, perceived_variance, behavioral_term
  real :: alpha, beta, gamma, demand
  real :: gross_return, turnover, cost_per_turnover, net_return

  expected_return = 0.08
  perceived_variance = 0.03
  behavioral_term = 0.40
  alpha = 1.2
  beta = 0.7
  gamma = 0.5

  demand = alpha * expected_return - beta * perceived_variance + gamma * behavioral_term

  gross_return = 0.05
  turnover = 1.4
  cost_per_turnover = 0.0025
  net_return = gross_return - cost_per_turnover * turnover

  print *, "Synthetic behavioral demand:", demand
  print *, "Synthetic net return after turnover:", net_return
end program behavioral_demand
